feat(table): expandable table demo

jenny-s51 opened the pull request:
<!-- What changes are being made? Please link the issue being addressed. -->
**What**: Closes #7271

patternfly-build commented:
Preview: https://patternfly-react-pr-7277.surge.sh

kmcfaul commented:
I think you're missing the isFullPage flag on the md as the example is still inline

mmenestr commented:
If I'm looking at the right demo: https://patternfly-react-pr-7277.surge.sh/components/table/react-demos/expandcollapse-all/ - is it normal that the first row doesn't have a checkbox?

tlabaj reviewed (changes requested):
I think we want this one to match core's demo.  The core demo has a toolbar and pagination at the top and bottom.

jenny-s51 commented:
Thank you @tlabaj - I've added top/bottom pagination to this PR to match the core demo. Per team consensus, the rest of the toolbar features will be determined and implemented once https://github.com/patternfly/patternfly-design/issues/1157 and https://github.com/patternfly/patternfly-react/issues/7384 are complete.

Nice catch @mmenestr - removed the first checkbox to match the core demo!

mcarrano reviewed (approved):
LGTM! Thanks @jenny-s51 .

mattnolting commented on packages/react-table/src/demos/table-demos/ExpandCollapseAll.jsx:
```diff
@@ -115,24 +126,53 @@ class ExpandCollapseAllTableDemo extends React.Component {
-        </Table>
+        <DashboardWrapper hasPageTemplateTitle>
+          <PageSection isWidthLimited>
+            {tableToolbar}
```
Needs to be wrapper with `Card`

mattnolting commented on packages/react-table/src/demos/table-demos/ExpandCollapseAll.jsx:
```diff
@@ -115,24 +126,53 @@ class ExpandCollapseAllTableDemo extends React.Component {
   render() {
     const { columns, rows, collapseAllAriaLabel } = this.state;
 
+    const tableToolbar = (
+      <Toolbar id="compact-toolbar">
+        <ToolbarContent>
+          <ToolbarItem variant="pagination">{this.renderPagination('top', true)}</ToolbarItem>
+        </ToolbarContent>
+      </Toolbar>
+    );
```
I think I'd remove the toolbar if it only has pagination as the white space on the left seems odd. @mcarrano wdyt?

tlabaj replied:
We could also add some other controls to the toolbar?

mcarrano replied:
I agree with @mattnolting . But I'm OK to either remove it or add some simple filter actions. But did we say we would remove these toolbars until we worked out the correct structure to match core, or am I confusing this with another issue @tlabaj ?

jenny-s51 replied:
+1 @mattnolting @mcarrano, I think it would make sense to remove the toolbar here as well since we need to figure out the correct table structure and data first before implementing controls that would use the table data

tlabaj replied:
That was a different PR @mcarrano but it applies here as well.  We can add the toolbar back once we have a data set.

mattnolting reviewed (commented):
Nice work! Left a few comments.

mmenestr commented:
Last thing - is it normal that the first row doesn't have an expansion icon?

jenny-s51 commented:
> Last thing - is it normal that the first row doesn't have an expansion icon?

Yes @mmenestr , the first row doesn't have an expansion icon because there currently are no designs or sample data for all the expanded states of https://pf4.patternfly.org/components/table/html-demos/expandable/

We can update this table to fully match the core demo once we have all the data/designs i.e. when
https://github.com/patternfly/patternfly-design/issues/1157 and https://github.com/patternfly/patternfly-react/issues/7384 are complete.

mattnolting reviewed (approved):
LPTM! Thanks @jenny-s51!

patternfly-build commented:
Your changes have been released in: 
 - eslint-plugin-patternfly-react@4.53.0
 - @patternfly/react-catalog-view-extension@4.65.0
 - @patternfly/react-charts@6.67.0
 - @patternfly/react-code-editor@4.55.0
 - @patternfly/react-console@4.65.0
 - @patternfly/react-core@4.214.0
 - @patternfly/react-docs@5.75.0
 - @patternfly/react-icons@4.65.0
 - @patternfly/react-inline-edit-extension@4.59.0
 - demo-app-ts@4.174.0
 - @patternfly/react-integration@4.176.0
 - @patternfly/react-log-viewer@4.59.0
 - @patternfly/react-styles@4.64.0
 - @patternfly/react-table@4.83.0
 - @patternfly/react-tokens@4.66.0
 - @patternfly/react-topology@4.61.0
 - @patternfly/react-virtualized-extension@4.61.0
 - transformer-cjs-imports@4.52.0

Thanks for your contribution! :tada:
